### PR TITLE
[MAINTENANCE] Add `suite_name` to `ExpectationSuiteValidationResult`

### PR DESCRIPTION
**Pull request by cdkini**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/core/expectation_validation_result.py [hunk 1]
```diff
@@ -503,13 +503,15 @@ class ExpectationSuiteValidationResult(SerializableDictDot):
...
+        result_url: A URL where the results are stored.
     """  # noqa: E501
 
     def __init__(  # noqa: PLR0913
         self,
-        success: Optional[bool] = None,
-        results: Optional[list] = None,
+        success: bool,
+        results: list[ExpectationValidationResult],
+        suite_name: str,
```
Comment (cdkini) [inline great_expectations/core/expectation_validation_result.py:514]: These should be guaranteed fields - what is a result if we don't have results, an origin suite, and a success metric?

File: great_expectations/core/expectation_validation_result.py [hunk 2]
```diff
@@ -654,11 +652,11 @@ def describe(self) -> str:
 class ExpectationSuiteValidationResultSchema(Schema):
     success = fields.Bool()
     results = fields.List(fields.Nested(ExpectationValidationResultSchema))
+    suite_name = fields.String(required=True, allow_none=False)
     evaluation_parameters = fields.Dict()
     statistics = fields.Dict()
     meta = fields.Dict(allow_none=True)
     id = fields.UUID(required=False, allow_none=True)
-    checkpoint_name = fields.String(required=False, allow_none=True)
```
Comment (cdkini) [inline great_expectations/core/expectation_validation_result.py:661]: This was wrong - it might be in meta but shouldn't be at the top level

File: great_expectations/core/expectation_validation_result.py [hunk 3]
```diff
@@ -613,7 +610,7 @@ def get_failed_validation_results(
     ) -> ExpectationSuiteValidationResult:
         validation_results = [result for result in self.results if not result.success]
 
-        successful_expectations = sum(exp.success for exp in validation_results)
+        successful_expectations = sum(exp.success or False for exp in validation_results)
```
Comment (joshua-stauffer) [inline great_expectations/core/expectation_validation_result.py:613]: why is this required? do some ValidationResults have a `None` value for `success`?

**Review by joshua-stauffer — APPROVED** [review]
looks reasonable 👍

File: great_expectations/core/expectation_validation_result.py (hunk 3 above)
Comment (cdkini) [reply great_expectations/core/expectation_validation_result.py:613]: yeah we'd need a separate effort to clean up the child `ExpectationValidationResult`. I'll follow up
